lesson21/index.js

```diff
@@ -110,5 +112,60 @@ const createTd = (usersData, keys) => {
   return fragment;
 }
 
+const createSortBtn = () => {
+  const sortBtn = createElementWithClassName("button", "sort-btn");
+  sortBtn.dataset.sortStatus = "standard";
+  sortBtn.id = "js-sort-btn";
+  return sortBtn;
+}
+
+const setClickInSortBtn = () => {
+  const standardRows = [...document.querySelector("tbody").querySelectorAll("tr")];
+  const sortBtn = document.getElementById("js-sort-btn");
+
+  sortBtn.addEventListener("click", (e) => {
+    let currentStatus = e.target.dataset.sortStatus;
+    currentStatus = switchSortStatus(currentStatus);
+    e.target.dataset.sortStatus = currentStatus;
+
+    const sortedRows = getSortedRows(currentStatus, standardRows);
+
+    const tbody = document.querySelector("tbody");
+    sortedRows.forEach((row) => {
+      tbody.appendChild(row);
+    })
+  });
+}
+
+const switchSortStatus = (status) => {
+  if (status === "standard") {
+    return "asc";
+  } else if (status === "desc") {
+    return "standard";
+  } else if (status === "asc") {
+    return "desc";
+  }
+}
+
+const getSortedRows = (status, standardRows) => {
+  if (status === "standard") {
+    return standardRows;
+  }
+  return SortById(status, standardRows);
+}
+
+const SortById = (status, standardRows) => {
```

Comment:
Thanks for the review.
Changed so that the sort function can be applied to other items as well.
The index of the clicked cell is now the argument, and the function name has been changed.
Please check it out.


c463746